### PR TITLE
add Spring.gitignore

### PR DESCRIPTION
**Pull request by hknochi**
add an ignore file for SpringFramework Projects

### CODE REVIEW

File: Spring.gitignore [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+buildSrc/build
+integration-repo
+.settings
+/spring-*/build
+.springBeans
+spring-test/test-output/
+/src/asciidoc/build
+target/
+
+# Eclipse artifacts, including WTP generated manifests
```
Comment (shiftkey) [inline Spring.gitignore]: We maintain the IDE-specific rules under the `Global` folder as this is easier to maintain when there are new releases to IDEs. Could you remove this section and the IDEA section below?

File: Spring.gitignore [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+*.iml
+*.ipr
+*.iws
+.idea
+out
+test-output
+atlassian-ide-plugin.xml
+.gradletasknamecache
+
+# Ignore some other files
```
Comment (shiftkey) [inline Spring.gitignore]: I'd love for these to be documented a bit so that others can understand why these rules exist...

**Comment by shiftkey** [comment]
I'm trying to wrap my head around where this should live:

 - aside from a few specific exceptions, templates at the root represent the defaults for programming languages
 - we use the `Global` folder for collecting IDEs and tools that build upon whatever you - as many of them support multiple languages

From reading the project site (apologies if I make mistakes here, please clarify if I'm wrong at all) Spring is a framework for building enterprise-scale Java apps. So my first instinct is to ask **"could this be done with rules in Java.gitignore?"** (like we do with other framework-esque relationships like Django is to Python) There's a lot of rules here, so perhaps that's not the ideal approach - others might not want or need these rules.

The next feeling that I have is that this is something you would opt-in for later - rather than when creating a new repo. Let's say your company is a Java shop, so when starting a new project you go and create the repository using the `Java.gitignore` template. A bit later, you work out that the team wants to use IntelliJ - so you add the `Global/JetBrains.gitignore` template to your own. **Does that "pick things as you go" flow feel like how you'd adopt `Spring.gitignore`?**

Of course `Global` is an amorphous namespace and doesn't quite feel right for this situation, but perhaps that's a decision from years ago that I need to revisit...

File: Spring.gitignore [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+# Ignore some other files 
+*.java.hsp
+*.sonarj
+*.sw*
+ivy-cache
+jxl.log
+jmx.log
+derby.log
+argfile*
+pom.xml
```
Comment (hdiwan) [inline Spring.gitignore]: Why is this line included? Generally, I want my build directive included in source control.

**Comment by shiftkey** [comment]
Closing due to inactivity and pending questions.
